src/codeview.js

```diff
@@ -329,6 +335,58 @@ var Codeview = GObject.registerClass({
         return GLib.SOURCE_REMOVE;
     }
 
+    // returns a diff in the form of an ed-script, which is the format output by
+    // the diff tool that takes the fewest bytes. It is forward-only; you cannot
+    // apply it in reverse.
+    async _diffFromLastCompiledText() {
```

Comment:
It's already compressed when being sent to the server. The reason for storing only a diff of each step is because I thought that these metrics would otherwise fill up the bucket quickly, as they take up more space on disk than most other events.

Reply:
Right, lets assume it also saves them compressed :)

Reply:
It doesn't ... they are stored in GVariant byte format